composer.json

```
@@ -42,6 +42,14 @@
       "Neighborhoods\\KojoWorkerDecoratorComponent\\": [
         "src",
         "fab"
+      ],
+      "Neighborhoods\\KojoWorkerDecoratorComponent\\WorkerV1\\": [
+        "WorkerV1\\src",
+        "WorkerV1\\fab"
+      ],
+      "Neighborhoods\\KojoWorkerDecoratorComponent\\WorkerDecorators\\CrashedThresholdV1\\": [
+        "WorkerDecorators\\CrashedThresholdV1\\src",
+        "WorkerDecorators\\CrashedThresholdV1\\fab"
```

I think my concern with this pattern is that composer.json and the buphalo script both have to be updated for every decorator added. I wonder if a structure like this might not make things more accessible / easier to contribute / less prone to mistakes: i.e. Something that is more along the lines of our Pit of Success value (easy to do the right thing and harder to do the wrong thing).

```suggestion
      "Neighborhoods\\KojoWorkerDecoratorComponent\\WorkerDecorators\\": [
        "WorkerDecorators\\src",
        "WorkerDecorators\\fab"
```

---

That would make it harder for users to add the needed paths if they want specific decorators only.
With the current approach it's
```
->addSourcePath('vendor/neighborhoods/kojo-worker-decorator-component/WorkerV1')
->addSourcePath('vendor/neighborhoods/kojo-worker-decorator-component/WorkerDecorators/CrashedThresholdV1')
->addSourcePath('vendor/neighborhoods/kojo-worker-decorator-component/WorkerDecorators/ExceptionHandlingDecoratorV1')
```

With the proposed change it would be
```
->addSourcePath('vendor/neighborhoods/kojo-worker-decorator-component/WorkerV1')
->addSourcePath('vendor/neighborhoods/kojo-worker-decorator-component/WorkerDecorators/fab/CrashedThresholdV1')
->addSourcePath('vendor/neighborhoods/kojo-worker-decorator-component/WorkerDecorators/src/CrashedThresholdV1')
->addSourcePath('vendor/neighborhoods/kojo-worker-decorator-component/WorkerDecorators/fab/ExceptionHandlingDecoratorV1')
->addSourcePath('vendor/neighborhoods/kojo-worker-decorator-component/WorkerDecorators/src/ExceptionHandlingDecoratorV1')
```

---

Yeah, it's definitely a tradeoff, but adding `src/<Component>` and `fab/<Component>` is already part of the standard process for every component you want to add to a container. And really, thinking about this more makes me feel like it might be easiest (and matches other existing patterns that we already use) to have the file structure look like this:

```
src/
    WorkerV1/
    WorkerDecorators/
fab/
    WorkerV1/
    WorkerDecorators/
```

(I'm probably creating a ListingService specific wrapper for the container builder, so I've been thinking about this kind of thing for a while, and maintaining the existing pattern feels like more of a Pit of Success than trying to come up with something clever here.)

---

(And yes, I realize I was the one that came up with the original idea of putting separate `src/` and `fab/` directories in each one, instead of vice versa, but after seeing it in practice, I like it less and less)

---

No problem.
Another difference in addition to the structure are the method names in the generated AwareTraits. When having the src folder in the root the method names are longest. With the approach I initially used they were super short.